### PR TITLE
Add Windows Build

### PR DESCRIPTION
**Pull request by RUIJIEZHONG66166**

### CODE REVIEW

File: .github/workflows/pull.yml [hunk 1]
```diff
@@ -35,6 +35,13 @@ jobs:
       abi: 0
       ut: op_extended
       runner: linux.idc.xpu
+  
+  preci-windows:
```
Comment (chuanqi129) [inline .github/workflows/pull.yml:39]:
```suggestion
  preci-windows-build:
```

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"
+          python -c "import torch; print(torch.__config__.show())"
+          python -c "import torch; print(torch.__config__.parallel_info())"
+          python -c "import torch; print(torch.__config__.torch.xpu.device_count())"
+
+      - name: Upload Windows binary
+        if: ${{ ! cancelled() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: Inductor-Windows-Binary-${{ github.event.pull_request.number || github.sha }}
```
Comment (chuanqi129) [inline .github/workflows/windows.yml:136]:
```suggestion
          name: Torch-XPU-Windows-Binary-${{ github.event.pull_request.number || github.sha }}
```

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+          )
+          if defined TORCH_WHL (
+            pip install %TORCH_WHL%
+            echo "INFO - torch whl installed"
+          ) else (
+            echo "INFO - No torch whl found."
+            exit /b 1
+          )
+          pip install --force-reinstall dist/*.whl
+          git clone https://github.com/pytorch/vision && cd vision && python setup.py install && cd ..
```
Comment (chuanqi129) [inline .github/workflows/windows.yml:120]: Let's remove torchvision part in torch xpu build

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+          python -c "import torch; print(torch.__config__.parallel_info())"
+          python -c "import torch; print(torch.__config__.torch.xpu.device_count())"
+
+      - name: Upload Windows binary
+        if: ${{ ! cancelled() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: Inductor-Windows-Binary-${{ github.event.pull_request.number || github.sha }}
+          path: ${{ github.workspace }}/../pytorch/dist
+
```
Comment (chuanqi129) [inline .github/workflows/windows.yml:138]: keep the last line without any space string

File: .github/scripts/install_oneapi_bundle.bat [hunk 5]
```diff
@@ -0,0 +1,56 @@
+REM Description: Install Intel Support Packages on Windows
+REM BKM reference: https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpu/2-5.html
+REM To-do: Add driver installation in this file.
+
+set ONEAPI_PARENT_DIR=C:\Program Files (x86)\Intel
+set INTEL_ONEAPI_PYTORCH_BUNDLE_URL=https://registrationcenter-download.intel.com/akdlm/IRC_NAS/5ca2021d-dd1a-4ab1-bd52-758fe63cf827/w_intel-for-pytorch-gpu-dev_p_0.5.2.19_offline.exe
```
Comment (chuanqi129) [inline .github/scripts/install_oneapi_bundle.bat:6]: Let's update the bundle link to 0.5.3 and install pti also

File: .github/scripts/install_oneapi_bundle.bat
Comment (chuanqi129) [inline .github/scripts/install_oneapi_bundle.bat:1]: rename to `install_xpu.bat`

File: .github/workflows/windows.yml [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        description: Runner label
+
+permissions: read-all
+
+env: 
+    USE_XPU: 1
+    VS2022INSTALLDIR: 'C:\Program Files\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\Build\vcvarsall.bat'
+
+jobs:
+  Torch-XPU-Windows-Tests:
```
Comment (chuanqi129) [inline .github/workflows/windows.yml:44]:
```suggestion
  Torch-XPU-Windows-Build:
```

File: .github/workflows/pull.yml (hunk 1 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/pull.yml]: Done

File: .github/workflows/windows.yml (hunk 2 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/windows.yml]: Done

File: .github/workflows/windows.yml (hunk 3 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/windows.yml]: Done

File: .github/workflows/windows.yml (hunk 4 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/windows.yml]: Done

File: .github/scripts/install_oneapi_bundle.bat
Comment (RUIJIEZHONG66166) [reply .github/scripts/install_oneapi_bundle.bat:1]: Done

File: .github/workflows/windows.yml (hunk 6 above)
Comment (RUIJIEZHONG66166) [reply .github/workflows/windows.yml]: Done

File: .github/scripts/install_oneapi_bundle.bat (hunk 5 above)
Comment (RUIJIEZHONG66166) [reply .github/scripts/install_oneapi_bundle.bat]: Updated the bundle link to 0.5.3. Then, I will try to add pti installation and sync with US val team about it.

Comment (chuanqi129) [reply .github/scripts/install_oneapi_bundle.bat]: I guess we don't need sync, just follow the latest BKC. Anyway, you can submit another PR for it.

Comment (RUIJIEZHONG66166) [reply .github/scripts/install_oneapi_bundle.bat]: Its just for effort saving. We do not need to do it for both side. I will also try to do it later with another PR for it.

**Pull request closed without merging**